src/actions/kloudio/kloudio.ts

```diff
@@ -0,0 +1,178 @@
+
+    try {
+        const newUri = API_URL.replace(/['"]+/g, "")
+        const lambdaResponse = await https.post({
+        url: newUri,
+        headers: {"Content-Type": "application/json"},
+        json: true,
+        body: data,
+         }).catch((_err) => {
+           const error = JSON.parse(_err)
+           winston.error("parsing error code" + error.emailCode)
+           winston.error(_err.toString())
+          })
+
+        if (!lambdaResponse.success || lambdaResponse.success === false) {
+          winston.info("lambda url resp is not sucess " + lambdaResponse)
+          response = { success: false, message: lambdaResponse.message }
+        } else {
+          response = { success: true, message: lambdaResponse.message }
+        }
+    } catch (e) {
+      winston.error("Inside catch statement" + e)
+      response = { success: false, message: e.message }
+    }
+    return new Hub.ActionResponse(response)
+  }
+
+  async form() {
+    const form = new Hub.ActionForm()
+    form.fields = [{
+      label: "API Key",
+      name: "apiKey",
+      required: true,
+      type: "string",
+    }, {
+      label: "Google Sheets URL",
+      name: "url",
+      required: true,
+      type: "string",
+    }]
+    return form
+  }
+
+  protected generateAnonymousId() {
+    return uuid.v4()
+  }
+
+}
+
+async function parseData(lookerData: any, names: any, labels: any) {
+    return new Promise<any>( async (resolve, reject) => {
+        try {
+            for (const row of lookerData) {
+                const tempA = []
+                for (const label of names) {
+                    if (row[label].rendered) {
+                        tempA.push(row[label].rendered)
+                    } else {
+                        tempA.push(row[label].value)
+                    }
+                }
+                labels.push(tempA)
+            }
+            return resolve(labels)
+        } catch (error) {
+            return reject(error)
+        }
+    })
+}
+
+async function getS3Url(fileName: any, url: any, token: any ) {
+
+  let errsResponse = { success: true, message:  "success"}
+  const comurl = url + fileName + "&apiKey=" + token
+  const apiURL = comurl.replace(/['"]+/g, "")
+  const s3UrlResponse = await https.get({
+    url: apiURL,
+    headers: { ContentType: "application/json"},
+     }).catch((_err) => {
+        errsResponse = { success: false, message:  _err.error}
+        winston.error(_err.toString())
+      })
+  if (errsResponse.success === false || !errsResponse.success) {
+    return errsResponse
+  } else {
+    return JSON.parse(s3UrlResponse)
+  }
+}
+
+async function uploadToS32(url: any, s3Data1: any) {
+
+  const santUrl = url.replace(/['"]+/g, "")
+  const response = https.put({
+    url: santUrl,
+    headers: {"Content-Type": "application/json"},
+    json: true,
+    body: s3Data1,
+     }).catch((_err) => { winston.error(_err.toString()) })
```

Comment:
`_err` is used so should it be `err` instead? the `_` is for parameters that aren't getting used

Reply:
ok this has been changed to "err".